OpenRA.Mods.Common/Traits/SupportPowers/SpawnActorPower.cs

```diff
@@ -47,33 +59,117 @@ public SpawnActorPower(Actor self, SpawnActorPowerInfo info)
-			}
+
+				if (info.LifeTime > -1)
+				{
+					actor.QueueActivity(new Wait(info.LifeTime));
+					actor.QueueActivity(new RemoveSelf());
+				}
+			});
+		}
+
+		public override void SelectTarget(Actor self, string order, SupportPowerManager manager)
+		{
+			Game.Sound.PlayToPlayer(SoundType.UI, manager.Self.Owner, Info.SelectTargetSound);
+			Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
+				Info.SelectTargetSpeechNotification, self.Owner.Faction.InternalName);
+			self.World.OrderGenerator = new SelectSpawnActorPowerTarget(order, manager, this, MouseButton.Left);
+		}
+
+		public bool Validate(World world, SpawnActorPowerInfo info, CPos cell)
+		{
+			if (!world.Map.Contains(cell))
+				return false;
+
+			if (!info.AllowUnderShroud && world.ShroudObscures(cell))
+				return false;
+
+			if (info.Terrain != null && !info.Terrain.Contains(world.Map.GetTerrainInfo(cell).Type))
+				return false;
+
+			return true;
+		}
+	}
+
+	public class SelectSpawnActorPowerTarget : OrderGenerator
+	{
+		readonly SpawnActorPower power;
+		readonly SpawnActorPowerInfo info;
+		readonly SupportPowerManager manager;
+		readonly string order;
+		readonly MouseButton expectedButton;
+
+		public string OrderKey { get { return order; } }
+
+		public SelectSpawnActorPowerTarget(string order, SupportPowerManager manager, SpawnActorPower power, MouseButton button)
+		{
+			// Clear selection if using Left-Click Orders
+			if (Game.Settings.Game.UseClassicMouseStyle)
+				manager.Self.World.Selection.Clear();
+
+			this.manager = manager;
+			this.power = power;
+			this.order = order;
+			expectedButton = button;
+
+			info = (SpawnActorPowerInfo)power.Info;
+		}
+
+		protected override IEnumerable<Order> OrderInner(World world, CPos cell, int2 worldPixel, MouseInput mi)
+		{
+			world.CancelInputMode();
+
+			if (!power.Validate(world, info, cell))
+				yield break;
+
+			if (mi.Button == expectedButton)
+				yield return new Order(order, manager.Self, Target.FromCell(world, cell), false) { SuppressVisualFeedback = true };
+		}
+
+		protected override void Tick(World world)
+		{
+			// Cancel the OG if we can't use the power
+			if (!manager.Powers.ContainsKey(order))
+				world.CancelInputMode();
+		}
+
+		protected override IEnumerable<IRenderable> Render(WorldRenderer wr, World world) { yield break; }
+		protected override IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world) { yield break; }
+		protected override IEnumerable<IRenderable> RenderAnnotations(WorldRenderer wr, World world) { yield break; }
+		protected override string GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi)
+		{
+			if (!world.Map.Contains(cell))
+				return info.BlockedCursor;
+
+			if (!info.AllowUnderShroud && world.ShroudObscures(cell))
+				return info.BlockedCursor;
+
+			if (info.Terrain != null && !info.Terrain.Contains(world.Map.GetTerrainInfo(cell).Type))
+				return info.BlockedCursor;
+
+			return info.Cursor;
```

Comment:
We should be able to simplify this to
```suggestion
			return power.Validate(world, info, cell) ? info.Cursor : info.BlockedCursor;
```
now.

Reply:
Yes,  you are correct.

Reply:
Bonus points to you.